apps/server/src/controllers/paymentsController/fundPersonalSavingsWallet.ts

```diff
@@ -0,0 +1,124 @@
+import { Response } from 'express'
+import { Op } from 'sequelize'
+import { HTTP_STATUS_CODE } from '../../constants'
+import { RequestExt } from '../../middlware/authorization/authentication'
+import Wallets, { WalletType, OwnerType } from '../../models/wallets'
+import { fundWalletSchema } from '../../utils/validators'
+import Transactions, {
+  transactionStatus,
+  transactionType,
+  action,
+} from '../../models/transactions'
+import { v4 } from 'uuid'
+
+export const fundPersonalSavingsWallet = async (
+  req: RequestExt,
+  res: Response
+) => {
+  try {
+    const { _userId: userId, ...rest } = req.body
+    const verityAmount = { amount: rest.amount }
+    const reqData = fundWalletSchema.strict().safeParse(verityAmount)
+
+    if (!reqData.success) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: reqData.error.issues,
+      })
+    }
+
+    const { amount } = reqData.data
+
+    const savingsWallet = await Wallets.findOne({
+      where: {
+        [Op.and]: [
+          { ownerId: userId },
+          { ownerType: OwnerType.USER },
+          { type: WalletType.SAVINGS },
+        ],
+      },
+    })
+    const globalWallet = await Wallets.findOne({
+      where: {
+        [Op.and]: [
+          { ownerId: userId },
+          { ownerType: OwnerType.USER },
+          { type: WalletType.GLOBAL },
+        ],
+      },
+    })
+
+    if (globalWallet && savingsWallet) {
+      if (globalWallet.totalAmount < amount) {
+        return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+          message: 'Insufficient funds in your global wallet',
+        })
+      }
+
+      if (amount <= 0) {
+        return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+          message: 'Invalid amount',
+        })
+      }
+
+      const newGlobalBalance = globalWallet.totalAmount - amount
+      const newSavingsBalance = savingsWallet.totalAmount + amount
+      const newSavingsIncome = savingsWallet.totalIncome + amount
+      const newIncome = {
+        date: new Date(),
+        amount,
+      }
+      const newEarnings = [...savingsWallet.earnings, newIncome]
```

Comment:
I think since we have our results saved to a variable, savingsWallet for instance

we can directly modify it, in which case:

`const newEarnings = savingsWallet.earnings.push(newIncome) ==>  Time Complexity is O(1)
`
vs

`const newEarnings = [...savingsWallet.earnings, newIncome] ==> Time Complexity is O(n)
`

WDYT?